src/libraries/Common/src/System/Net/Http/Http3/QPack/QPackEncodingException.cs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+#if KESTREL
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http3.QPack
+#else
+namespace System.Net.Http.QPack
+#endif
+{
+    internal sealed class QPackEncodingException : Exception
```

Comment:
If this exception could bubble out to the user, it should be [Serializable] and expose a deserialization ctor.